Even if EDITOR is not set, mutt will always use /usr/bin/editor (which 
is set by update-alternatives), rather than falling back to vi

--- a/init.c
+++ b/init.c
@@ -3114,7 +3114,7 @@ void mutt_init (int skip_sys_rc, LIST *c
   {
     p = getenv ("EDITOR");
     if (!p)
-      p = "vi";
+      p = "/usr/bin/editor";
   }
   Editor = safe_strdup (p);
   Visual = safe_strdup (p);
--- a/init.h
+++ b/init.h
@@ -589,7 +589,7 @@ struct option_t MuttVars[] = {
   ** .pp
   ** This variable specifies which editor is used by mutt.
   ** It defaults to the value of the \fC$$$VISUAL\fP, or \fC$$$EDITOR\fP, environment
-  ** variable, or to the string ``vi'' if neither of those are set.
+  ** variable, or to the string ``/usr/bin/editor'' if neither of those are set.
   */
   { "encode_from",	DT_BOOL, R_NONE, OPTENCODEFROM, 0 },
   /*
